python_modules/dagster/dagster/_core/definitions/events.py

```diff
@@ -189,6 +190,24 @@ def from_coercible_or_definition(
         else:
             return AssetKey.from_coercible(arg)
 
+    @staticmethod
+    def from_coercible_to_asset_dep(arg: "CoercibleToAssetDep") -> "AssetKey":
```

Comment:
From Nick 
> Name should be from_coercible_to_asset_key?

https://github.com/dagster-io/dagster/pull/16432#discussion_r1323535829

Reply:
https://github.com/dagster-io/dagster/pull/16432#discussion_r1323544586

ah ok i was wondering if this name would be confusing, and it is. my intention was "from CoercibleToAssetDep" not "from Coercible to AssetDep".

The existing "coercible" methods are:
* `from_coercible` which does `CoercibleToAssetKey` -> `AssetKey`
* `from_coercible_or_definition` which does `CoercibleToAssetKey | AssetsDefinition | SourceAsset` -> `AssetKey`

I'm unsure how to keep consistency with this pattern, while not introducing confusion with the method name. i'll ponder and respond with ideas

Reply:
cc @schrockn  moved the convo about naming here

Reply:
Maybe the answer is that it just shouldn't be a static method on this class and we should force people to do

`AssetDep.from_coercible(value).asset_key`

Reply:
resolution: we'll move this function to a private method of `AssetDep`